api/files.py

```diff
@@ -238,3 +238,33 @@ def _save_multipart_files(self, dest_path, hash_alg):
                         'size': os.path.getsize(os.path.join(dest_path, filename)),
                         'mimetype': util.guess_mimetype(filename)
                     }, os.path.join(dest_path, filename))
+
+
+FILE_EXTENSIONS = {
+    "archive":      [ ".zip", ".tbz2", ".tar.gz", ".tbz", ".tar.bz2", ".tgz", ".tar", ".txz", ".tar.xz" ],
+    "document":     [ ".docx", ".doc" ],
+    "image":        [ ".jpg", ".tif", ".jpeg", ".gif", ".bmp", ".png", ".tiff" ],
+    "pdf":          [ ".pdf" ],
+    "presentation": [ ".ppt", ".pptx" ],
+    "source code":  [ ".c", ".py", ".cpp", ".js", ".m", ".json", ".java" ],
+    "spredsheet":   [ ".xls", ".xlsx" ],
+    "tabular data": [ ".csv.gz", ".csv" ],
+    "text":         [ ".txt" ],
+    "video":        [ ".mpeg", ".mpg", ".mov", ".mp4" ],
+
+    "qa":           [ ".qa.png", ".qa.json" ],
+
+    "bval":         [ ".bval" ],
+    "bvec":         [ ".bvec" ],
+    "dicom":        [ ".dcm" ],
+    "gephysio":     [ ".gephysio.zip" ],
+    "nifti":        [ ".nii.gz", ".nii" ],
+    "pfile":        [ ".7.gz", ".7" ]
+}
+
+def guess_type_from_filename(filename):
+    for key in FILE_EXTENSIONS:
+        for extension in FILE_EXTENSIONS[key]:
+            if filename.endswith(extension):
+                return key
+    return None
```

Comment:
This may not work as intended with "double" extensions. `foo.gephysio.zip` may come back as `archive` or `gephysio`. Compare with [alternative implementation in reaper](https://github.com/scitran/reaper/blob/master/reaper/folder_uploader.py#L24-L38).